app/views/i18n/I18NEditModelView.coffee

```diff
@@ -128,54 +140,55 @@ module.exports = class I18NEditModelView extends RootView
     #- Enable patch submit button
 
     @$el.find('#patch-submit').attr('disabled', null)
+    @madeChanges = true
 
   onLanguageSelectChanged: (e) ->
+    if @madeChanges
+      return unless confirm('You have unsaved changes!')
```

Comment:
I'd change to "You have unsaved changes! Really discard them?", to make it more clear what "OK" and "Cancel" would do.